sqlbrite-contentprovider/src/main/java/com/squareup/sqlbrite_contentprovider/SqlBriteContentProvider.java

```diff
@@ -0,0 +1,230 @@
+   * notifications for when the supplied {@code uri}'s data changes. Unsubscribe when you no longer
+   * want updates to a query.
+   * <p>
+   * <b>Warning:</b> this method does not perform the query! Only by subscribing to the returned
+   * {@link Observable} will the operation occur.
+   *
+   * @see ContentResolver#query(Uri, String[], String, String[], String)
+   * @see ContentResolver#registerContentObserver(Uri, boolean, ContentObserver)
+   */
+  public Observable<Query> createQuery(@NonNull final Uri uri, @Nullable final String[] projection,
+      @Nullable final String selection, @Nullable final String[] selectionArgs, @Nullable
+  final String sortOrder, final boolean notifyForDescendents) {
+    final Query query = new Query() {
+      @Override public Cursor run() {
+        return query(uri, projection, selection, selectionArgs, sortOrder);
+      }
+    };
+    return Observable.create(new Observable.OnSubscribe<Query>() {
+      @Override public void call(final Subscriber<? super Query> subscriber) {
+        contentResolver.registerContentObserver(uri, notifyForDescendents,
+            new ContentObserver(contentObserverHandler) {
+              @Override public void onChange(boolean selfChange) {
+                if (subscriber.isUnsubscribed()) {
+                  contentResolver.unregisterContentObserver(this);
+                } else {
+                  subscriber.onNext(query);
+                }
+              }
+            });
+      }
+    }).startWith(query);
+  }
+
+  /**
+   * Query the given URI, returning a {@link Cursor} over the result set.
+   * <p>
+   * For best performance, the caller should follow these guidelines:
+   * <ul>
+   * <li>Provide an explicit projection, to prevent
+   * reading data from storage that aren't going to be used.</li>
+   * <li>Use question mark parameter markers such as 'phone=?' instead of
+   * explicit values in the {@code selection} parameter, so that queries
+   * that differ only by those values will be recognized as the same
+   * for caching purposes.</li>
+   * </ul>
+   * </p>
+   *
+   * @see ContentResolver#query(Uri, String[], String, String[], String)
+   */
+  public final Cursor query(@NonNull Uri uri, @Nullable String[] projection,
+      @Nullable String selection, @Nullable String[] selectionArgs, @Nullable String sortOrder) {
+    if (logging) {
+      log("QUERY\n  uri: %s\n  projection: %s\n  selection: %s\n  selectionArgs: %s\n  sortOrder: %s",
+          uri.toString(), Arrays.toString(projection), selection, Arrays.toString(selectionArgs),
+          sortOrder);
+    }
+    return contentResolver.query(uri, projection, selection, selectionArgs, sortOrder);
+  }
+
+  /**
+   * Inserts a row into a table at the given URL.
+   *
+   * If the content provider supports transactions the insertion will be atomic.
+   *
+   * @see ContentResolver#insert(Uri, ContentValues)
+   */
+  public final Uri insert(@NonNull Uri uri, @NonNull ContentValues values) {
```

Comment:
the only real save is that you dont have to carry two separate objects around for your db operations but its likely better to just use the contentresolver regardless - ill rm these